tests/test_cli.py

```diff
@@ -140,6 +140,86 @@ def test_create_user_config() -> None:
     }
 
 
+@pytest.mark.parametrize(
+    ("user_config_args,config_type,str_value"),
+    [
+        ('{"description":"test","network":"0.0.0.0/0"}', "array", [{"description": "test", "network": "0.0.0.0/0"}]),
+        ("0.0.0.0,0.0.0/8", "array", ["0.0.0.0", "0.0.0/8"]),
+        ("123", "integer", 123),
+        ("10.0", "number", 10.0),
+        ("false", "boolean", False),
+    ],
+)
+def test_convert_str_to_value(user_config_args: str, config_type: str, str_value: Union[str, dict]) -> None:
```

Comment:
I think `str_value` is not a great name for the expected value when it can be of other types too, and it seems it can also be an array contrary to what the type hint states. I personally would be happy with `expected_value: Any` in a test context, but if you feel like being more pedantic then that's fine too.